### PR TITLE
Adds fireplaces (cozy as fuck)

### PR DESCRIPTION
**Pull request by Niknakflak**
Simple preview, there's other overlays of the fire being smaller and of some embers, but it mostly looks the same.

---

Fireplaces are now added to the game for mappers to use.  They take up to 4 logs and slowly burn themselves out.  Comes complete with 3 directions (south, east and west) but only the south one has the overlays for obvious reasons.  The east and west directions look fine but are really for if you have no other choice with a map, they are however still fully functional.  

Code is some of my own, some written/input from Joan, oranges, Xhuis, Mrperson, and various other people who helped me in this miserable experience.

Sprite is a recolored and added directions that originated from /vg/ but was modified by myself as well as Nienhaus. (He did recolors/new directions, I redid where the overlays sat and moved the sprite around within the 64x64 area).  

/vg/ pull request is here: https://github.com/d3athrow/vgstation13/pull/7820

I may have fucked up some maps, we'll see and I'll fix later. Jesus christ.

### CODE REVIEW

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 1]
```diff
@@ -19,56 +19,156 @@
...
 
 
+#define SECONDS_PER_LOG 150 //How many seconds a log will burn for
 
+//Fireplaces//
 
-/*Cabin code*/
 /obj/structure/fireplace
 	name = "fireplace"
+	desc = "A large stone fireplace, warm and cozy"
```
Comment (Bawhoppen) [inline code/modules/awaymissions/mission_code/Cabin.dm]: its definitely brick in the vg image

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 2]
```diff
@@ -19,56 +19,156 @@
...
+/obj/structure/fireplace/proc/try_light(obj/item/O, mob/user)
+	if (!wood)
+		user << "<span class='warning'>[src] needs some wood to burn!</span>"
+		return FALSE
+	if (lit == 1)
+		user << "<span class='warning'>It's already lit!</span>"
+		return FALSE
+
+	var/lighting_text = "<span class='notice'>[user] lights the [src] with the [O].</span>"
+	if(istype(O, /obj/item/weapon/weldingtool))
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: no.
use ```is_hot()```

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 3]
```diff
@@ -19,56 +19,156 @@
...
+	pixel_x = -16
+	var/wood = 0
+	var/lit = 0
+
+
+/obj/structure/fireplace/proc/try_light(obj/item/O, mob/user)
+	if (!wood)
+		user << "<span class='warning'>[src] needs some wood to burn!</span>"
+		return FALSE
+	if (lit == 1)
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```if(lit)```

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 4]
```diff
@@ -19,56 +19,156 @@
...
+		lighting_text = "<span class='notice'>[user] lights the [src] with the [O]. What a badass. </span>"
+	else if(istype(O, /obj/item/weapon/lighter/greyscale))
+		lighting_text = "<span class='notice'>After some fiddling, [user] manages to light the [src] with [O].</span>"
+	else if(istype(O, /obj/item/weapon/lighter))
+		lighting_text =  "<span class='rose'>With a single flick of their wrist, [user] smoothly lights the [src] with [O]. Classy.</span>"
+	else if(istype(O, /obj/item/weapon/melee/energy))
+		lighting_text = "<span class='warning'>[user] swings their [O], lighting the [src] in the proccess.</span>"
+	if(O.is_hot())
+		visible_message(lighting_text)
+		lit = 1
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```TRUE``` not 1

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 5]
```diff
@@ -19,56 +19,156 @@
...
 /obj/structure/fireplace
 	name = "fireplace"
+	desc = "A large stone fireplace, warm and cozy"
+	icon = 'icons/obj/fireplace.dmi'
+	icon_state = "fireplace"
+	density = 0
+	anchored = 1
+	pixel_x = -16
+	var/wood = 0
+	var/lit = 0
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```FALSE``` not 0

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 6]
```diff
@@ -19,56 +19,156 @@
...
+		add_overlay("fireplace_glow")
+
+
+/obj/structure/fireplace/process()
+	wood --
+	update_icon()
+	if(wood > 1)
+		playsound(src, 'sound/effects/comfyfire.ogg',50,0, 0, 1)
+	else if(!wood)
+		lit = 0
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```FALSE``` not 0

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 7]
```diff
@@ -19,56 +19,156 @@
...
+		if(400 to 600)
+			add_overlay("fireplace_fire4")
+			SetLuminosity(6)
+
+	if (wood != 0)
+		add_overlay("fireplace_glow")
+
+
+/obj/structure/fireplace/process()
+	wood --
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Don't put a space between ```wood``` and ```--```

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 8]
```diff
@@ -19,56 +19,156 @@
...
+			add_overlay("fireplace_fire2")
+			SetLuminosity(3)
+		if(300 to 400)
+			add_overlay("fireplace_fire3")
+			SetLuminosity(4)
+		if(400 to 600)
+			add_overlay("fireplace_fire4")
+			SetLuminosity(6)
+
+	if (wood != 0)
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```if(wood)``` please

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 9]
```diff
@@ -19,56 +19,156 @@
 
 
 
+#define SECONDS_PER_LOG 150 //How many seconds a log will burn for
```
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: ```SECONDS_PER_LOG``` with a value in deciseconds?
just call it ```TIME_PER_LOG``` or ```LOG_BURN_TIME``` or ```LOG_FUEL_TIME```

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 1 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: >someone didn't read

Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: .

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 5 above)
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: #lit

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 10]
```diff
@@ -19,56 +19,156 @@
...
-/*Cabin code*/
 /obj/structure/fireplace
 	name = "fireplace"
+	desc = "A large stone fireplace, warm and cozy"
+	icon = 'icons/obj/fireplace.dmi'
+	icon_state = "fireplace"
+	density = 0
+	anchored = 1
+	pixel_x = -16
+	var/wood = 0
```
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: nah

make this accept any burnable object

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: but then it'll always be true

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 11]
```diff
@@ -19,56 +19,156 @@
...
+
+	var/lighting_text = "<span class='notice'>[user] lights the [src] with the [O].</span>"
+	if(istype(O, /obj/item/weapon/weldingtool))
+		lighting_text = "<span class='notice'>[user] lights the [src] with the [O]. What a badass. </span>"
+	else if(istype(O, /obj/item/weapon/lighter/greyscale))
+		lighting_text = "<span class='notice'>After some fiddling, [user] manages to light the [src] with [O].</span>"
+	else if(istype(O, /obj/item/weapon/lighter))
+		lighting_text =  "<span class='rose'>With a single flick of their wrist, [user] smoothly lights the [src] with [O]. Classy.</span>"
+	else if(istype(O, /obj/item/weapon/melee/energy))
+		lighting_text = "<span class='warning'>[user] swings their [O], lighting the [src] in the proccess.</span>"
```
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: i think this is standardized

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 12]
```diff
@@ -19,56 +19,156 @@
...
+	if(O.is_hot())
+		visible_message(lighting_text)
+		lit = 1
+		//IT'S LIT FAM
+		START_PROCESSING(SSobj, src)
+		return TRUE
+
+/obj/structure/fireplace/attackby(obj/item/T, mob/user)
+	if(istype(T,/obj/item/stack/sheet/mineral/wood))
```
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: change it to accept any burnable object

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 13]
```diff
@@ -19,56 +19,156 @@
...
+		if(wood > 4)
+			user << "<span class = 'warning'>There's already enough logs in the [src].</span>"
+			return
+		var/woodnumber = input(user, "Fireplace Fuel: Max 4 logs.", "How much wood do you want to add?", 0) as num //Something is causing this to break
+		woodnumber = Clamp(woodnumber,0,4)
+		var/obj/item/stack/sheet/mineral/wood/woody = T
```
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: nice

**Comment by Iamgoofball** [comment]
https://github.com/Niknakflak/-tg-station/pull/3 made a fix PR

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 10 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: No

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 12 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: No

**Comment by Niknakflak** [comment]
Fuck off goofball

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 12 above)
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Can you allow throwing in paper then, for burning info in the fireplace?

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 13 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: /vg/ var port name

Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: nice x2

**Comment by coiax** [comment]
we already have fireplaces in away mission code, but they are not as pretty and as featureful as this :+1:

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 11 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: yeah, should probably make this a proc or defines or something, copy paste is bad

**Comment by coiax** [comment]
OH this is replacing that fireplace code. Even more :+1:

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 7 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: 1) Using process() ticks is inaccurate, should use a world.time timer.
2) Use `hotspot_expose` on the turf that the fireplace is in while it's lit.

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: This is a copypaste of the cig lighting code

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 14]
```diff
@@ -19,56 +19,156 @@
...
+		//IT'S LIT FAM
+		START_PROCESSING(SSobj, src)
+		return TRUE
+
+/obj/structure/fireplace/attackby(obj/item/T, mob/user)
+	if(istype(T,/obj/item/stack/sheet/mineral/wood))
+		if(wood > 4)
+			user << "<span class = 'warning'>There's already enough logs in the [src].</span>"
+			return
+		var/woodnumber = input(user, "Fireplace Fuel: Max 4 logs.", "How much wood do you want to add?", 0) as num //Something is causing this to break
```
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Please no input menus for throwing wood on a fireplace. Just take as many as possible, wasting the rest. It's a fireplace, it doesn't have a nice friendly input syste,

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 15]
```diff
@@ -19,56 +19,156 @@
...
+	else if(try_light(T,user))
+		return
+
+/obj/structure/fireplace/update_icon()
+	cut_overlays()
+	luminosity = 0
+	switch(wood)
+		if(0 to 100)
+			add_overlay("fireplace_fire0")
+			SetLuminosity(1)
```
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Adjusting the luminosity during update_icon... :/

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 14 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: No, that's stupid, you can't fit fifty logs in a fireplace.  Ignoring this

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 15 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Thats how it works dum dum

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 7 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Not gonna give it a hotspot

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 11 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Dont really see the reason too

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (Niknakflak) [inline code/modules/awaymissions/mission_code/Cabin.dm]: So is /vg/s, it works fine

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 15 above)
Comment (Iamgoofball) [inline code/modules/awaymissions/mission_code/Cabin.dm]: We try to ensure only icon shit happens in update_icon, move it

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/modules/awaymissions/mission_code/Cabin.dm]: is_hot() is used, this is just for special lighting texts.

**Comment by xxalpha** [comment]
fuck as cozy

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: The fact that a copy paste works doesn't mean it's a good idea, don't be lazy.

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 7 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: It's a fireplace on fire. Hotspot expose doesn't make a hotspot, it ignites flammable gasses, which is used by lighters, candles, welding torches, sparks

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 14 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Then just take as many as possible, don't use input() boxes unless REALLY needed.

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 8 above)
Comment (optimumtact) [inline code/modules/awaymissions/mission_code/Cabin.dm]: got wood? :grinning:

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 12 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Burning paper is a good idea, given that we're actually just incrementing a timer . Can give it a smaller BURN_VALUE to compensate

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: oh this is *JUST* for messages? that's even worse tbh, if it was for the functionality it'd be wrong still but this is just extra wrong.

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 11 above)
Comment (coiax) [inline code/modules/awaymissions/mission_code/Cabin.dm]: > copy paste is bad

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 14 above)
Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Cabin.dm]: the idea is to take up to 4 in from the stack, NikNak, the amount you take would be ```max(4-woodnumber, 0)```, no need for a popup.

Comment (MrPerson) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Gotta agree that using an input box is really unnecessary and ugly looking all around. Just use as many as possible up to the max the fireplace can hold. IE what sheets do.

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 4 above)
Comment (Xhuis) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Does it really matter? I think it's personal preference at this point

Comment (xxalpha) [inline code/modules/awaymissions/mission_code/Cabin.dm]: i got yelled at for suggesting TRUE and FALSE :confused:

**Comment by Niknakflak** [comment]
If @coiax could stop fucking interacting with me, that'd be fantastic

**Comment by Xhuis** [comment]
Can both of you grow up?

**Comment by coiax** [comment]
Don't look at me, I'm not the one who's refusing to make required changes to a PR because I suggested them. If and when Niknak makes a valid suggestion on one of my PRs, I'll be implementing that right away.

**Comment by Niknakflak** [comment]
sorry, I have you set on ignore V(._.)V

**Comment by Xhuis** [comment]
Case in point.

**Pull request closed without merging**